core/domain/blog_services.py

```diff
@@ -798,20 +820,17 @@ def get_published_blog_post_summaries(
         max_limit = size
     else:
         max_limit = feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_HOMEPAGE
-    blog_post_rights_models: Sequence[blog_models.BlogPostRightsModel] = (
-        blog_models.BlogPostRightsModel.query(
-            blog_models.BlogPostRightsModel.blog_post_is_published == True  # pylint: disable=singleton-comparison
+    blog_post_summary_models: Sequence[blog_models.BlogPostSummaryModel] = (
+        blog_models.BlogPostSummaryModel.query(
+            blog_models.BlogPostSummaryModel.published_on != None  # pylint: disable=singleton-comparison, inequality-with-none, line-too-long
```

Comment:
SInce the statement is inside `.query()` it does not work with is not None. Changing != to is not None give the following error on running the query : 
![Screenshot from 2023-02-07 17-11-08](https://user-images.githubusercontent.com/68547101/217333612-356f1df8-f0cb-4db6-b5c9-15222e114f20.png)

Reply:
Done